src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -529,6 +529,22 @@ private sealed class Mapper : MapperBase
                         if (typeValueCount % valCount != 0)
                             throw Contracts.Except($"Input shape mismatch: Input '{_parent.Inputs[i]}' has shape {originalShape.ToString()}, but input data is of length {typeValueCount}.");
 
+                        //This cover the 2-variable senario e.g. [?,?,?,3] where we can assume typeDims provides the information of [W, H, C]
+                        var originalShapeDims = originalShape.dims;
+                        var originalShapeNdim = originalShape.ndim;
+                        if (numOfUnkDim == colTypeDims.Length && originalShapeNdim == numOfUnkDim + 1 && colTypeDims.Length == 3)
+                        {
```

Comment:
nit: It appears that you are checking that numOfUnkDim and colTypeDims.Length must both be 3. It is more readable to compare them explicitly against 3.

Reply:
sure